main.go

```diff
@@ -170,6 +170,12 @@ func main() {
 		}
 	}
 
+	listenerPullPolicy := os.Getenv("CONTROLLER_MANAGER_LISTENER_IMAGE_PULL_POLICY")
+	switch pullPolicy := corev1.PullPolicy(listenerPullPolicy); pullPolicy {
+	case corev1.PullAlways, corev1.PullNever, corev1.PullIfNotPresent:
+		actionsgithubcom.SetListenerImagePullPolicy(pullPolicy)
+	}
```

Comment:
feels like a little bit of duplication on checking the `case` from both places?

Reply:
Yes for sure
I left it here just in case we want to add logging maybe? Do you think we should trace?
If yes, we can add 
1. default case: "Using default image pull policy for AutoscalingListener"
1. On line 176: Setting AutoscalingListener image pull policy to: %q

Reply:
i think we should trace it in `main.go` and remove the `switch/case` in resourceBuilder?

Reply:
How do you feel about accepting a string and returning a bool from SetListenerImagePullPolicy?
So set can control what values are okay, and main can know if the configuration is actually applied or we are using defaults

Reply:
I just wanted to have safety that `actionsgithubcom` package decides what values are correct and can be applied. It is already driving the logic for almost everything, so it feels to me like this logic belongs there